extension/chrome/settings/modules/my_key_update.ts

```diff
@@ -6,79 +6,96 @@ import { Catch } from '../../../js/common/platform/catch.js';
 import { Store } from '../../../js/common/platform/store.js';
 import { Ui } from '../../../js/common/browser.js';
 import { Settings } from '../../../js/common/settings.js';
-import { Pgp } from '../../../js/common/core/pgp.js';
+import { Pgp, KeyInfo } from '../../../js/common/core/pgp.js';
 import { Lang } from '../../../js/common/lang.js';
 import { Assert } from '../../../js/common/assert.js';
 import { Attester } from '../../../js/common/api/attester.js';
 import { Api } from '../../../js/common/api/api.js';
 import { Url } from '../../../js/common/core/common.js';
+import { View } from '../../../js/common/view.js';
 
 declare const openpgp: typeof OpenPGP;
 
-Catch.try(async () => {
+View.run(class MyKeyUpdateView extends View {
+  private readonly acctEmail: string;
+  private readonly showKeyUrl: string;
+  private readonly primariKiPromise: Promise<KeyInfo>;
+  private readonly inputPrivateKey = $('.input_private_key');
+  private readonly prvHeaders = Pgp.armor.headers('privateKey');
 
-  const uncheckedUrlParams = Url.parse(['acctEmail', 'longid', 'parentTabId']);
-  const acctEmail = Assert.urlParamRequire.string(uncheckedUrlParams, 'acctEmail');
-  const longid = Assert.urlParamRequire.optionalString(uncheckedUrlParams, 'longid') || 'primary';
-  const showKeyUrl = Url.create('my_key.htm', uncheckedUrlParams);
+  constructor() {
+    super();
+    const uncheckedUrlParams = Url.parse(['acctEmail', 'longid', 'parentTabId']);
+    const acctEmail = Assert.urlParamRequire.string(uncheckedUrlParams, 'acctEmail');
+    const longid = Assert.urlParamRequire.optionalString(uncheckedUrlParams, 'longid') || 'primary';
+    this.acctEmail = acctEmail;
+    this.showKeyUrl = Url.create('my_key.htm', uncheckedUrlParams);
+    this.primariKiPromise = (async () => (await Store.keysGet(acctEmail, [longid]))[0])();
```

Comment:
Just put this into `render`, for simplicity

Reply:
I also was thinking about it but if I put it into `render` method the type of `primariKi` will be `KeyInfo | undefined`

Reply:
I'd force it with `!`, I think that's a cleaner solution, although not perfect.